client/rest-high-level/src/main/java/org/elasticsearch/client/ml/datafeed/DatafeedUpdate.java

```diff
@@ -287,6 +287,7 @@ public Builder(DatafeedUpdate config) {
             this.delayedDataCheckConfig = config.delayedDataCheckConfig;
         }
 
+        @Deprecated
```

Comment:
I think this annotation should be in 7.4 and the method removed entirely in 8.x.

When we decided to allow `job_id` to still be specified as long as it matched the current `job_id` this was for the get-modify locally-update workflow.  But that only makes sense if some code is taking a whole datafeed config from the get API and sending it back with modifications to the update API.  This method is setting the `job_id` individually, and if somebody is doing that they might as well delete the call from their code when upgrading to 8.x.